client.go

```diff
@@ -419,9 +424,12 @@ func (c *Client) stream(method, path string, streamOptions streamOptions) error
 		}
 		clientconn := httputil.NewClientConn(dial, nil)
```

Comment:
It is said that ClientConn is low-level and old(which `NewClientConn` func returns). Do you mind fixing this btw :)
http://golang.org/pkg/net/http/httputil/#ClientConn